core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/impl/RawInfraImpl.kt

```diff
@@ -106,33 +115,66 @@ class RawInfraImpl(
     val logicalSignalPool: StaticPool<LogicalSignal, LogicalSignalDescriptor>,
     val physicalSignalPool: StaticPool<PhysicalSignal, PhysicalSignalDescriptor>,
     val zonePathPool: StaticPool<ZonePath, ZonePathDescriptor>,
-    val zonePathMap: Map<ZonePathSpec, ZonePathId>
+    val zonePathMap: Map<ZonePathSpec, ZonePathId>,
+    val operationalPointPartPool: StaticPool<OperationalPointPart, OperationalPointPartDescriptor>,
+    val trackSectionIdMap: Map<String, TrackSectionId>,
 ) : RawInfra {
     override val trackNodes: StaticIdxSpace<TrackNode>
         get() = trackNodePool.space()
     override val trackSections: StaticIdxSpace<TrackSection>
         get() = trackSectionPool.space()
 
-    override fun getTrackSectionDetectors(trackSection: TrackSectionId): StaticIdxList<Detector> {
-        return trackSectionPool[trackSection].detectors
-    }
-
     override fun getTrackSectionId(trackSection: TrackSectionId): String {
         return trackSectionPool[trackSection].name
     }
 
+    override fun getTrackSectionFromId(id: String): TrackSectionId? {
+        return trackSectionIdMap[id]
+    }
+
     override fun getTrackSectionChunks(trackSection: TrackSectionId): StaticIdxList<TrackChunk> {
         return trackSectionPool[trackSection].chunks
     }
 
+    override fun getDirTrackSectionChunks(trackSection: TrackSectionId, direction: Direction):
+            DirStaticIdxList<TrackChunk> {
+        var chunks = trackSectionPool[trackSection].chunks
+        if (direction == Direction.DECREASING)
+            chunks = chunks.reversed()
+        val res = mutableDirStaticIdxArrayListOf<TrackChunk>()
+        for (chunk in chunks)
+            res.add(DirTrackChunkId(chunk, direction))
+        return res
+    }
```

Comment:
I don't think this API needs to be directional: you could directly return `trackSectionPool[trackSection].chunks` and make an extension method to iterate on chunks with a given direction. This way, there's no need to make a copy the chunk list on each call